coreax/solvers/coresubset.py

```diff
@@ -458,3 +486,301 @@ def selection_function(i: int, _kernel_similarity_penalty: ArrayLike) -> Array:
             self.unroll,
         )
         return refined_coreset, solver_state
+
+
+def _greedy_cmmd_loss(
+    random_key: KeyArrayLike,
+    candidate_coresets: Array,
+    feature_gramian: Array,
+    response_gramian: Array,
+    training_cme: Array,
+    regularisation_parameter: float,
+    identity: Array,
+    inverse_approximator: Optional[RegularisedInverseApproximator] = None,
+) -> Array:
+    """
+    Given an array of candidate coreset indices, compute the CMMD loss for each.
+
+    Primarily intended for use with :class`GreedyCMMD.
+
+    :param random_key: Key for random number generation
+    :param candidate_coresets: Array of indices representing all possible "next"
+        coresets
+    :param feature_gramian: Feature kernel gramian
+    :param response_gramian: Response kernel gramian
+    :param training_cme: CME evaluated at all possible pairs of training data
+    :param regularisation_parameter: Regularisation parameter for stable inversion of
+        array, negative values will be converted to positive
+    :param identity: Block "identity" matrix
+    :param inverse_approximator: Instance of
+        :class:`coreax.inverses.RegularisedInverseApproximator`, default value of
+        :data:`None` uses :class:`coreax.inverses.LeastSquareApproximator`
+
+    :return: GreedyCMMD loss for each candidate coreset
+    """
+    # Extract all the possible "next" coreset arrays
+    extract_indices = (candidate_coresets[:, :, None], candidate_coresets[:, None, :])
+    coreset_feature_gramians = feature_gramian[extract_indices]
+    coreset_response_gramians = response_gramian[extract_indices]
+    coreset_cmes = training_cme[extract_indices]
+
+    # Invert the coreset feature gramians
+    if inverse_approximator is None:
+        inverse_approximator = LeastSquareApproximator(random_key)
```

Comment:
This line is not hit by a unit test

Reply:
Removed default value as this is not meant to be used by users and also got rid of the `__post_init__` as pyrght complains about it constantly and it is easy to get around it